### PR TITLE
[RA2 Ch4] Re-addition of host OS requirements

### LINKED ISSUE #2798: [RA2 Ch4] Lost Host OS Requirements
There was a clause of host OS requirements in the previous release of RA2 (e.g., https://github.com/cntt-n/CNTT/blob/Snezka/doc/ref_arch/kubernetes/chapters/chapter04.md) capturing the requirements necessary for CNF and host OS interoperability.
The clause is removed in the following releases (perhaps due to re-structing), and there are no similar requirements in the current Ch4 now.
We think the original requirements (e.g., minimum version of linux kernel) are important for CNFs to run across different container infrastructure, and therefore should be added back into the RA2 spec.

### PR DESCRIPTION
**Pull request by ulikleber**
Resolves #2798 .

This re-enters Host OS requirements which were lost after Snezka.

### CODE REVIEW

File: doc/ref_arch/kubernetes/chapters/chapter04.rst [hunk 1]
```diff
@@ -32,31 +32,191 @@ virtual machine and an installed Operating System. In order for a Kubernetes Nod
...
-============== ============================================== ===================================================================================================================================================================================================================================================================================================================== ========================================================================================================================================================================================== =====================================================================================================
++--------------+--------------+----------------------------------------------+-----------------------+-----------------------+
```
Comment (CsatariGergely) [inline doc/ref_arch/kubernetes/chapters/chapter04.rst:35]: This is a duplicate of #2912

File: doc/ref_arch/kubernetes/chapters/chapter04.rst [hunk 2]
```diff
@@ -32,31 +32,191 @@ virtual machine and an installed Operating System. In order for a Kubernetes Nod
...
 
 **Table 4-1:** Node Specifications
 
+Node Operating System
+---------------------
+
+In order for a Host OS to be compliant with this Reference Architecture it must meet the following requirements:
+
++--------------+--------------+----------------------------------------------+-----------------------+-----------------------+
```
Comment (CsatariGergely) [inline doc/ref_arch/kubernetes/chapters/chapter04.rst:166]: Please leave a space before and after the borders of the table. This is needed for proper rst rendering.

**Review by CsatariGergely — CHANGES_REQUESTED** [review]

File: doc/ref_arch/kubernetes/chapters/chapter04.rst (hunk 1 above)
Comment (ulikleber) [reply doc/ref_arch/kubernetes/chapters/chapter04.rst:35]: Ah, didn't notice between all the line-wrap PRs... 
Please help me how to fix this.

File: doc/ref_arch/kubernetes/chapters/chapter04.rst [hunk 3]
```diff
@@ -32,31 +32,191 @@ virtual machine and an installed Operating System. In order for a Kubernetes Nod
...
+|              |              |chosen as the baseline because kubeadm is     |                       |                       |
+|              |              |focussed on installing and managing the       |                       |                       |
+|              |              |lifecycle of Kubernetes and nothing else,     |                       |                       |
+|              |              |hence it is easily integrated into            |                       |                       |
+|              |              |higher-level and more complete tooling for    |                       |                       |
+|              |              |the full lifecycle management of the          |                       |                       |
+|              |              |infrastructure, cluster add-ons, etc.         |                       |                       |
++--------------+--------------+----------------------------------------------+-----------------------+-----------------------+
+|``ra2.os.003``|Windows Server|Windows Server 2019 (this can be used for     |tbd                    |tbd                    |
```
Comment (rgstori) [inline doc/ref_arch/kubernetes/chapters/chapter04.rst:184]: why not any version of Windows Server?

File: doc/ref_arch/kubernetes/chapters/chapter04.rst [hunk 4]
```diff
@@ -32,31 +32,191 @@ virtual machine and an installed Operating System. In order for a Kubernetes Nod
...
+|              |Deployment    |supports req.lcm.gen.01 (automated            |                       |                       |
+|              |              |deployments)                                  |                       |                       |
++--------------+--------------+----------------------------------------------+-----------------------+-----------------------+
+
+**Table 4-2:** Operating System Requirements
+
+Table 4-3 lists the Linux kernel versions that comply with this Reference Architecture specification.
+
++------------+------------------+--------------------------+
+|OS Family   |Version(s)        |Notes                     |
```
Comment (rgstori) [inline doc/ref_arch/kubernetes/chapters/chapter04.rst:209]: change to "Kernel Version"

File: doc/ref_arch/kubernetes/chapters/chapter04.rst [hunk 5]
```diff
@@ -32,31 +32,191 @@ virtual machine and an installed Operating System. In order for a Kubernetes Nod
...
++============+==================+==========================+
+|Linux       |3.10+             |                          |
```
Comment (rgstori) [inline doc/ref_arch/kubernetes/chapters/chapter04.rst:211]: probably needs updating

File: doc/ref_arch/kubernetes/chapters/chapter02.rst [hunk 6]
```diff
@@ -211,6 +217,9 @@ Cloud Infrastructure Software Profile Capabilities
 
 **(2)** In Kubernetes based infrastructures packet monitoring is out of the scope for the infrastructure.
 
+**(3)** Host operating system must provide drivers etc. to support listed capabilities. More information,
```
Comment (rgstori) [inline doc/ref_arch/kubernetes/chapters/chapter02.rst:220]: can be moved to third column of `i.os.001`

File: doc/ref_arch/kubernetes/chapters/chapter04.rst (hunk 1 above)
Comment (ulikleber) [reply doc/ref_arch/kubernetes/chapters/chapter04.rst:35]: Waiting for #2912 to be merged.

File: doc/ref_arch/kubernetes/chapters/chapter04.rst (hunk 2 above)
Comment (ulikleber) [reply doc/ref_arch/kubernetes/chapters/chapter04.rst]: done.

File: doc/ref_arch/kubernetes/chapters/chapter04.rst (hunk 3 above)
Comment (ulikleber) [reply doc/ref_arch/kubernetes/chapters/chapter04.rst]: Removed "2019"

File: doc/ref_arch/kubernetes/chapters/chapter04.rst (hunk 4 above)
Comment (ulikleber) [reply doc/ref_arch/kubernetes/chapters/chapter04.rst]: done

File: doc/ref_arch/kubernetes/chapters/chapter02.rst (hunk 6 above)
Comment (ulikleber) [reply doc/ref_arch/kubernetes/chapters/chapter02.rst]: done

**Review by CsatariGergely — APPROVED** [review]

**Review by rgstori — APPROVED** [review]

**Review by walterkozlowski — APPROVED** [review]

**Pull request merged**